packages/flutter/lib/src/painting/_network_image_io.dart

```diff
@@ -81,39 +70,112 @@ class NetworkImage extends image_provider.ImageProvider<image_provider.NetworkIm
+        // initialized [SendPort]), then just send download request down to it.
+        _requestPort.send(downloadRequest);
+      } else {
+        if (_pendingLoader == null) {
+          // If worker isolate creation was not started, start creation now.
+          _pendingLoadRequests = <_PendingLoadRequest>[];
+
+
+
+          _pendingLoader = _setupIsolate()..then((Isolate isolate) {
+              final RawReceivePort handleError = RawReceivePort((List<String> errorAndStackTrace) {
+                _cleanupDueToError(errorAndStackTrace[0]);
+              });
+              isolate.addErrorListener(handleError.sendPort);
+              isolate.resume(isolate.pauseCapability);
+            }).catchError((dynamic error, StackTrace stackTrace) {
+              _cleanupDueToError(error);
+            });
+        }
+        // Record download request so it can either send a request when isolate is ready or handle errors.
+        _pendingLoadRequests.add(_PendingLoadRequest(
+            (SendPort sendPort) { sendPort.send(downloadRequest); },
+            (dynamic error) { downloadRequest.sendPort.send(_DownloadResponse.error(error.toString())); }
+        ));
+      }
+
+      final TransferableTypedData transferable = await bytesCompleter.future;
+      downloadResponseHandler.close();
+
+      final Uint8List bytes = transferable.materialize().asUint8List();
+      if (bytes.isEmpty)
         throw Exception('NetworkImage is an empty file: $resolved');
 
-      return PaintingBinding.instance.instantiateImageCodec(bytes);
+      return await PaintingBinding.instance.instantiateImageCodec(bytes);
     } finally {
       chunkEvents.close();
     }
   }
 
+  void _cleanupDueToError(dynamic error) {
+    for (_PendingLoadRequest request in _pendingLoadRequests) {
+      request.handleError(error);
+    }
+    _pendingLoadRequests = null;
+    _pendingLoader = null;
+  }
+
+
+  Future<Isolate> _setupIsolate() {
+    // This is used to get _requestPort [SendPort] that can be used to
+    // communicate with worker isolate: when isolate is spawned it will send
+    // it's [SendPort] over via this [RawReceivePort].
+    // Received [sendPort] can also be [null], which indicates that worker
+    // isolate exited after being idle.
+    final RawReceivePort receivePort = RawReceivePort((SendPort sendPort) {
+      _requestPort = sendPort;
+      if (sendPort == null) {
+        _pendingLoader = null;
+        assert(_pendingLoadRequests.isEmpty);
+        return;
+      }
+
+      // When we received [SendPort] for the worker isolate, we send all
+      // pending requests that were accumulated before worker isolate provided
+      // it's port (before [_requestPort] was populated).
+      for (_PendingLoadRequest pendingRequest in _pendingLoadRequests) {
+        // [sendPort] being null indicates that worker has been idle and exited.
+        // That should not happen if there are pending download requests.
+        assert(sendPort != null);
```

Comment:
This assert can move outside the for loop.

Reply:
Okay, I removed the `assert` because we have `if (sendPort == null)` just before the loop.